app/helpers/application_helper/button/auth_key_pair_cloud_create.rb

```diff
@@ -2,11 +2,9 @@ class ApplicationHelper::Button::AuthKeyPairCloudCreate < ApplicationHelper::But
   def disabled?
     # check that at least one EMS the user has access to supports
     # creation or import of key pairs.
-    Rbac.filtered(ManageIQ::Providers::CloudManager.all).each do |ems|
-      if ems.supports?(:auth_key_pair_create)
-        return false
-      end
-    end
+    supported_types = ManageIQ::Providers::CloudManager::AuthKeyPair.descendants.select { |klass| klass.supports?(:create) }.map(&:module_parent).map(&:name)
+    return false if Rbac.filtered(ManageIQ::Providers::CloudManager.where(:type => supported_types)).any?
```

Comment:
(not making any formatting suggestions, just formatted so I could read the comment w/o scroll)

```ruby
supported_types = ManageIQ::Providers::CloudManager::AuthKeyPair.ems_types_supporting(:create)
if Rbac.filtered(ManageIQ::Providers::CloudManager.where(:type => supported_types)).any?
  return false
end
```

I like the way you have it, and the PR I put in will support your code. but wondering if there is a more elegant way to do it.

```ruby
Rbac.filtered(ManageIQ::Providers::CloudManager.types_supports(:create, by_child: "AuthKeyPair")).any?
Rbac.filtered(ManageIQ::Providers::CloudManager.types_supports("AuthKeyPair" => :create)).any?
```

Reply:
Oh 100% there definitely is a better way to do this if we add some helper methods to core